compiler/rustc_mir/src/borrow_check/type_check/mod.rs

```diff
@@ -2742,10 +2742,10 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
         category: ConstraintCategory,
     ) {
         self.prove_predicates(
-            Some(ty::PredicateKind::Trait(
-                ty::TraitPredicate { trait_ref },
-                hir::Constness::NotConst,
-            )),
+            Some(ty::PredicateKind::Trait(ty::TraitPredicate {
+                trait_ref,
+                constness: hir::Constness::NotConst,
```

Comment:
we could probably use the const context of the item currently being type checked to decide on `Const` here.

Reply:
This is only used to prove that a lang item is implemented e.g. `CoerceUnsized`, `Sized`, and `Copy`